### PR TITLE
HMM-Lib: Bump Java version to 8

### PR DESCRIPTION
**Pull request by boldtrn**
We switched to Junit5 some time ago. Junit5 requires Java 8. The hmm-lib is still on Java 7. 

When you try to execute the `ViterbiAlgorithmTest` in InteliJ, there is a build error, as the pom sets the language level to Java 7.

```
/graphopper/graphhopper/hmm-lib/src/test/java/com/bmw/hmm/ViterbiAlgorithmTest.java:153:9
java: cannot access java.util.function.BooleanSupplier
  class file for java.util.function.BooleanSupplier not found
```

Bumping the version to Java 8 fixes that problem. Running through the command line works fine, so I am a bit unsure about this change, but I guess bumping the version should not hurt?

### CODE REVIEW

**Comment by easbar** [comment]
Hm, I can run the test in IntelliJ without a problem. But I also think it is/should be no problem to upgrade to Java 8

File: hmm-lib/pom.xml [hunk 1]
```diff
@@ -64,8 +64,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
```
Comment (easbar) [inline hmm-lib/pom.xml:64]: Can you just remove the entire plugin, please? It will be taken from parent pom then.

**Comment by boldtrn** [comment]
Yes I always thought it might be related to my setup, but I double checked on a second device and I saw the same result. I use the Maven integration of IntelliJ.

I updated the PR to remove the compiler plugin :)

**Comment by easbar** [comment]
Thanks

**Pull request merged**